src/test/kotlin/org/jetbrains/gradle/ext/BuildIdeArtifactTest.kt

```diff
@@ -5,343 +5,362 @@ import junit.framework.Assert.assertFalse
 
-    val srcFile = myProject.layout.projectDirectory
-            .dir(dirName)
-            .file(fileName).asFile
+        srcFile.parentFile.mkdir()
+        srcFile.writeText(payload)
 
-    srcFile.parentFile.mkdir()
-    srcFile.writeText(payload)
+        ideArtifact.artifact.directoryContent(dirName)
+        ideArtifact.createArtifact()
 
-    ideArtifact.artifact.directoryContent(dirName)
-    ideArtifact.createArtifact()
+        val target = myProject.layout.buildDirectory
+                .dir(DEFAULT_DESTINATION).get()
+                .dir(artifactName)
+                .file(fileName).asFile
 
-    val target = myProject.layout.buildDirectory
-            .dir(DEFAULT_DESTINATION).get()
-            .dir(artifactName)
-            .file(fileName).asFile
+        assertThat(target).exists()
+                .hasContent(payload)
+    }
 
-    assertThat(target).exists()
-            .hasContent(payload)
-  }
+    @Test
+    fun `test archive unpacking`() {
+        val payload = "Payload ;)"
+        val temp = createTempDir()
+        val fileName = "some.text"
 
-  @Test fun `test archive unpacking`() {
-    val payload = "Payload ;)"
-    val temp = createTempDir()
-    val fileName = "some.text"
+        try {
+            val testFile = File(temp, fileName)
+            testFile.writeText(payload)
+            val testArchive = myProject.layout.projectDirectory.file("archive.zip").asFile
+            createTestArchive(temp, testArchive)
 
-    try {
-      val testFile = File(temp, fileName)
-      testFile.writeText(payload)
-      val testArchive = myProject.layout.projectDirectory.file("archive.zip").asFile
-      createTestArchive(temp, testArchive)
+            ideArtifact.artifact.extractedDirectory(testArchive)
+            ideArtifact.createArtifact()
 
-      ideArtifact.artifact.extractedDirectory(testArchive)
-      ideArtifact.createArtifact()
+            val target = myProject.layout.buildDirectory
+                    .dir(DEFAULT_DESTINATION).get()
+                    .dir(artifactName)
+                    .file(fileName).asFile
 
-      val target = myProject.layout.buildDirectory
-              .dir(DEFAULT_DESTINATION).get()
-              .dir(artifactName)
-              .file(fileName).asFile
+            assertThat(target).exists()
+                    .hasContent(payload)
 
-      assertThat(target).exists()
-              .hasContent(payload)
+        } finally {
+            temp.deleteRecursively()
+        }
+    }
 
-    } finally {
-      temp.deleteRecursively()
+    fun createTestArchive(sourceDir: File, destinationFile: File) {
+        var out = ZipOutputStream(BufferedOutputStream(FileOutputStream(destinationFile)))
+        out.use { usedZipStream ->
+            sourceDir.walkBottomUp().forEach { file ->
+                if (!file.isDirectory()) {
+                    var entry = ZipEntry(file.absolutePath.removePrefix(sourceDir.absolutePath + "/"))
```

Comment:
Please, replace `sourceDir.absolutePath + "/"` with `sourceDir.absolutePath + File.separator` to fix test problem on Windows

Reply:
Fixed.